src/format-types/highlight/index.js

```diff
@@ -118,17 +118,19 @@ const HighlightButton = props => {
 
 	// Close the window if the user clicks outside.
 	const clickOutsideListener = useCallback( event => {
-		if ( isOpen ) {
-			if ( ! isElementDescendant( popoverEl.current, event.target ) && ! event.target.closest( '.components-popover' ) ) {
-				setIsOpen( false )
-			}
+		const outside = ! isElementDescendant( popoverEl.current, event.target )
+		const icon = event.target.closest(
+			'.stackable-components-toolbar__highlight'
+		)
+		if ( isOpen && outside && ! icon ) {
```

Comment:
Might be better if we use a more descriptive `isToolbarButton` instead of just `icon` as the variable name